File: packages/engine/python/get_db_schema.py
import os
import sys
import json
import asyncio

from typing import Optional

from py_logger import PyLogger
from database_client import DatabaseClient

logger = PyLogger(__name__)

async def main(
    dialect: Optional[str] = None,
    file_path: Optional[str] = None,
):
    host = os.getenv("DATABASE_CLIENT_HOST")
    port = os.getenv("DATABASE_CLIENT_PORT")
    user = os.getenv("DATABASE_CLIENT_USER")
    password = os.getenv("DATABASE_CLIENT_PASSWORD")
    database = os.getenv("DATABASE_CLIENT_DATABASE")
    schema = os.getenv("DATABASE_CLIENT_SCHEMA")

    db_client = DatabaseClient(dialect, host, port, user, password, database, schema, file_path)

    tables_metadata = db_client.get_db_schema()

    return tables_metadata

if __name__ == "__main__":
    try:
        if len(sys.argv) < 2:
            logger.error({"status": "error", "message": "No args provided"})
            sys.exit(1)
        scriptArgs = json.loads(sys.argv[1])
        logger.info({ "status": None, "message": "Script started!", "args": scriptArgs })
        result = asyncio.run(main(**scriptArgs))
        logger.warning(result)
        logger.info({"status": "success", "message": "Script ended!"})
        sys.exit(0)
    except Exception as e:
        logger.error({"status": "error", "message": str(e)})
        sys.exit(1)

File: packages/engine/src/python/_legacy/get-db-schema.ts
import { python } from '@trigger.dev/python'
import { env } from '@workspace/env'
import type { TableMetadata } from '../../lib/types'
import { pyArgs, pyPath, pyStreamingResult } from '../utils'

type GetDbSchemaArgs = {
  dialect?: string
  file_path?: string
}

type GetDbSchemaEnvVars = {
  host?: string
  port?: string
  user?: string
  password?: string
  database?: string
  schema?: string
}

type GetDbSchemaResult = {
  tables_metadata: TableMetadata[]
}

export async function getDbSchema(
  args: GetDbSchemaArgs,
  { envVars }: { envVars?: GetDbSchemaEnvVars } = {},
): Promise<GetDbSchemaResult> {
  const streamingResult = python.stream.runScript(
    pyPath('get_db_schema'),
    pyArgs(args),
    {
      env: {
        PYTHON_ENV: env.NODE_ENV,
        DATABASE_CLIENT_HOST: envVars?.host,
        DATABASE_CLIENT_PORT: envVars?.port,
        DATABASE_CLIENT_USER: envVars?.user,
        DATABASE_CLIENT_PASSWORD: envVars?.password,
        DATABASE_CLIENT_DATABASE: envVars?.database,
        DATABASE_CLIENT_SCHEMA: envVars?.schema,
      },
    },
  )

  const result = await pyStreamingResult<GetDbSchemaResult>(streamingResult)

  return result
}
